data_portal_api.yml

```diff
@@ -0,0 +1,332 @@
+openapi: 3.0.0
+info:
+  version: "1.0.0"
+  title: Corpora Data Portal API
+  description: >
+    This is the API for the Corpora Data Portal
+servers:
+  - description: production
+    url: https://api.corpora.czi-single-cell.science
+paths:
+  /v1/submissions:
+    get:
+      summary: List submissions
+      tags:
+        - submissions
+      description: >-
+        This lists all submissions and their UUIDs that currently exist in corpora. If a parameter is specified as a filter, then only projects that meet the status criteria will be outputted.
+
+        If user_uuid is provided, submissions from that user are returned. If no user uuid is specified the id of the authenticated user is used.
+      parameters:
+        - name: status
+          description: The status of the submissions to list.
+          in: query
+          schema:
+            type: string
+            enum: [open, closed]
+        - $ref: '#/components/parameters/user_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+    post:
+      summary: Create a new submission
+      tags:
+        - submissions
+      description: >-
+        opens a new submission and an empty project for editing. Executing this call will create a new entry in both the Submission and Project tables with empty fields and creates a new folder with the given project uuid in the data portal S3 bucket. If successful, a message will be returned with the project’s UUID.
+      responses:
+        '201':
+          description: A new submissions has been created.
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+  /v1/submissions/{submission_uuid}:
+    get:
+      summary: Get project submission by uuid
+      tags:
+        - submissions
+      description: >-
+        This call will return all available metadata information about a project’s submission, including datasets that are attached to the project.
+      parameters:
+        - $ref: '#/components/parameters/submission_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          description: Submission not found
```

Comment:
This one has a specific description that doesn't refer to the reference?

Reply:
I thought you wanted them more specific where applicable. We can keep it generic.

Reply:
Apologies for being unclear.. I was thinking more on the lines of:

`description: Submission not found
  schema: 
    $ref: '#/components/responses/404'
`

Reply:
But no worries - I think you make a good point that more of the error details can be in the message.